platform-common/src/main/java/com/softicar/platform/common/core/retry/AbstractRetrier.java

```diff
@@ -77,23 +78,40 @@ public T setRetryCount(int retryCount) {
 	}
 
 	/**
-	 * Defines the number of milliseconds to wait after a failed execution.
+	 * Defines the {@link Duration} to wait after a failed execution, before
+	 * trying again.
 	 *
-	 * @param delayMillis
-	 *            delay in milliseconds (the default is 0)
+	 * @param retryDelay
+	 *            the re-try delay; the default is {@link Duration#ZERO} (never
+	 *            <i>null</i> and never negative)
 	 * @throws IllegalArgumentException
 	 *             if the delay is negative
 	 * @return this object
 	 */
-	public T setRetryDelayMillis(long delayMillis) {
+	public T setRetryDelay(Duration retryDelay) {
 
-		if (delayMillis < 0) {
-			throw new IllegalArgumentException(String.format("Illegal number of %s millisecond for the re-try delay.", delayMillis));
+		if (retryDelay.isNegative()) {
+			throw new IllegalArgumentException(String.format("The re-try delay may not be negative."));
 		}
+		this.retryDelay = retryDelay;
+		return getThis();
+	}
 
-		this.delayMillis = delayMillis;
+	/**
+	 * Defines the number of milliseconds to wait after a failed execution,
+	 * before trying again.
+	 *
+	 * @param delayMillis
+	 *            the delay in milliseconds; the default is 0 (never negative)
+	 * @throws IllegalArgumentException
+	 *             if the delay is negative
+	 * @return this object
+	 * @deprecated use {@link #setRetryDelay(Duration)}
+	 */
+	@Deprecated
```

Comment:
I think this method should be removed instead. There is no known payload code that uses it.